packages/input/src/index.vue

```diff
@@ -270,18 +272,18 @@ export default defineComponent({
     })
     const isWordLimitVisible = computed(() => {
       return props.showWordLimit &&
-        ctx.attrs.maxlength &&
+        props.maxlength &&
         (props.type === 'text' || props.type === 'textarea') &&
         !inputDisabled.value &&
         !props.readonly &&
         !props.showPassword
     })
     const textLength = computed(() => {
-      return typeof props.modelValue === 'number' ? String(props.modelValue).length : (props.modelValue || '').length
+      return Array.from(nativeInputValue.value).length
     })
     const inputExceed = computed(() => {
       // show exceed style if length of initial value greater then maxlength
-      return isWordLimitVisible.value && (textLength.value > upperLimit.value)
+      return isWordLimitVisible.value && (textLength.value > props.maxlength)
```

Comment:
I found a potential bug though, the type of `maxlength` says it can be both `string` and `number` type, I think yo need to parse `maxlength` before comparison operation. WDYT?

Reply:
It is my opinion that 

- The maxlength only allows the number type  
```js
    maxlength: {
      type: Number,
    },
```
- `return isWordLimitVisible.value && (textLength.value > Number(props.maxlength) )`  Compatible with past string types